### PR TITLE
Cppify

### PR DESCRIPTION
**Pull request by thrimbor**
Several relatively small changes. Makes sure that NevolutionX builds properly against nxdk master, enables linking against libc++ (corrects #28 not linking), mounts the drives for compatibility with upcoming nxdk changes, tosses out folderparse and replaces it with winapi calls, plus some refactoring.

Please test that the file listing changes work - they should, but I wasn't able to properly test this yet because NevolutionX is crashy on my Xbox.

### CODE REVIEW

File: Includes/outputLine.cpp [hunk 1]
```diff
@@ -1,5 +1,13 @@
 #include "outputLine.h"
 
+#include <cstdarg>
+#include <cstdio>
+#include <SDL.h>
```
Comment (dracc) [inline Includes/outputLine.cpp]: No need for SDL.h if building with NXDK, or am I missing something?

File: Includes/subsystems.cpp [hunk 2]
```diff
@@ -1,16 +1,47 @@
...
+    outputLine("Mounting error: Could not mount drive C\n");
+  }
+  if (!nxMountDrive('E', "\\Device\\Harddisk0\\Partition1")) {
+    outputLine("Mounting error: Could not mount drive E\n");
+  }
+  if (!nxMountDrive('F', "\\Device\\Harddisk0\\Partition6")) {
+    outputLine("Mounting error: Could not mount drive F\n");
+  }
+  if (!nxMountDrive('G', "\\Device\\Harddisk0\\Partition7")) {
+    outputLine("Mounting error: Could not mount drive G\n");
```
Comment (dracc) [inline Includes/subsystems.cpp]: Should this really be reported as an error, G is highly optional IIRC?

**Comment by dracc** [comment]
I decided to steal half of your Makefile changes in the meantime. Thanks and sorry.

File: Includes/findXBE.cpp [hunk 3]
```diff
@@ -1,31 +1,37 @@
 #include "findXBE.h"
 #include <algorithm>
+#include <cstdio>
+#include <windows.h>
```
Comment (dracc) [inline Includes/findXBE.cpp]: This `#include` is NXDK specific - will (probably(?)) break compilation under Linux (apart from other issues such as an out-of-date CMakeLists.txt ...)

File: Includes/outputLine.cpp (hunk 1 above)
Comment (thrimbor) [reply Includes/outputLine.cpp]: Right, changed.

File: Includes/subsystems.cpp (hunk 2 above)
Comment (thrimbor) [reply Includes/subsystems.cpp]: I think it's fine for now, at least for logging purposes. It wouldn't present it to the user as an error, though.

File: Includes/findXBE.cpp (hunk 3 above)
Comment (thrimbor) [reply Includes/findXBE.cpp]: Fixed.

**Comment by dracc** [comment]
Great stuff. Thanks!

**Pull request merged**